framework/src/modules/interoperability/mainchain/endpoint.ts

```diff
@@ -25,4 +32,27 @@ export class MainchainInteroperabilityEndpoint extends BaseInteroperabilityEndpo
 			fee: MIN_RETURN_FEE_PER_BYTE_BEDDOWS.toString(),
 		};
 	}
+
+	public async isChainNameAvailable(context: ModuleEndpointContext): Promise<ResultJSON> {
```

Comment:
We can also add request/response schema and add it to metadata in the module class